farmerbot/parser/validator_test.go

```diff
@@ -0,0 +1,103 @@
+package parser
+
+import (
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/threefoldtech/tfgrid-sdk-go/farmerbot/internal"
+	"github.com/threefoldtech/tfgrid-sdk-go/farmerbot/mocks"
+)
+
+func TestValidateInput(t *testing.T) {
+	config := internal.Config{FarmID: uint32(25)}
+	ctrl := gomock.NewController(t)
+	mockGetNodes := mocks.NewMockSubstrate(ctrl)
+	mockGetNodes.EXPECT().GetNodes(config.FarmID).Times(13).Return([]uint32{20, 21, 22, 23, 24, 30, 31, 32, 34, 40, 41}, nil)
+	t.Run("test valid include, exclude, priority nodes and never shutdown nodes", func(t *testing.T) {
```

Comment:
Let's refactor to use e.g Table Driven Tests instead e.g
![image](https://github.com/user-attachments/assets/d87a82bf-8759-479c-a302-56676ea6bb58)